modules/cli/src/main/scala/scala/cli/commands/default/LegacyScalaOptions.scala

```diff
@@ -0,0 +1,62 @@
+package scala.cli.commands.default
+
+import caseapp.*
+import caseapp.core.Indexed
+
+import scala.build.Logger
+import scala.cli.commands.package0.Package
+import scala.cli.commands.tags
+
+/** Options covering backwards compatibility with the old scala runner.
+  */
+// format: off
+case class LegacyScalaOptions(
+  @Group("Scala")
+  @HelpMessage("Ignored legacy option. Deprecated equivalent of running a subsequent `package` command.")
```

Comment:
Is package by default in the scala runner actually?

Reply:
By default package command create a lightweight launcher JAR, like the "bootstrap" JAR files [generated by coursier](https://get-coursier.io/docs/cli-bootstrap#bootstraps). So this is different than what the save option generates.

I think scala runner generates Library JARs or fat JARs, @Gedochao could you check that?

Reply:
Good catch @tgodzik, it seems `package` is not part of the `scala` runner. For now, I suggest changing the help message to not mention it.

Reply:
@tgodzik ugh, y're right, `package` isn't included in the `scala` runner... are we ok with mentioning it with the `--power` command here?
I don't think we can deliver creating the jar here without it.

@lwronski ha, this is tricky. It seems we'd have to fix the `package` sub-command a bit to be compatible, as we don't fill all the required things in the manifest. I wasn't able to run a jar coming from `scala-cli package` as things are. And yes, we'd need to pass `--library` or `--assembly` flags anyway, as the bootstrap wouldn't do.

And even then, this functionality is actually slightly borked in the old `scala` runner already, as even `dotty`'s `scala` runner `refers to the wrong compiler version in the manifest...  So I have rising doubts if it's still useful to anyone

Reply:
@julienrf thoughts on mentioning `package` with the `--power` flag in the warning? Would it be acceptable when run as `scala`? That's the only sub-command capable of generating jars, even though it's not been accepted in the SIP

Reply:
Yes, I think that’s a good idea. But `--power` is not recognized (0.1.18). Is it still under implementation?

Reply:
it should be recognised in `0.1.18`.
Did you pass it as a launcher option?
It has to be passed before the sub-command to work.
```bash
▶ scala --power package .  
```

Reply:
anyhow, this will require some fixes in the `package` command, as we don't set the `Scala-Compiler-Version` and `Class-Path` fields in the jar manifest... (plus the `Main-Class` field is only available to be added for fat jars).
I'll create a separate issue for those and do it in a separate PR, I guess

Reply:
Ah, right, I didn’t pass `--power` _before_ `package`. Now it is recognized. But the option `--power` is not mentioned when I run `scala --help-full`. Is this intended?

Reply:
...yeah, I think it should be mentioned...
All of the launcher options (like `--cli-version` and `--power`) seem to be skipped from help, since they're passed before a subcommand is invoked (including the default sub-command).
That's another bug, it seems 😅

Reply:
the side-issues were extracted to #1698 & #1699.
The rest should already be addressed.